Д/з №5 (Четыркина Лена)

lenabatik opened the pull request:
Добавлен файл main.py с кодом автоматизации теста.
И приложен тест-кейс (в файле test_keys.md)

lenabatik committed:
Dobavleni files main.py (kod) i test_keys.md (opisanie test-kejsa)

skyline-gleb commented on main.py:
```diff
@@ -0,0 +1,44 @@
+        button_status = self.driver.find_element_by_name('submit')
+        text_button_status = button_status.get_attribute('class')
+
+        self.assertEqual("disabled", text_button_status)
```
здесь более красиво будет использование метода что предоставляют биндинги селениума `button_status.is_enabled()`, только ассерт будет немного другой, но это не проблема ведь

skyline-gleb commented on main.py:
```diff
@@ -0,0 +1,44 @@
+class Status_Button_Disabled_Test(TestCase):
+
+    def setUp(self):
+        self.driver = webdriver.Firefox() #создание класса
```
убери комментарий `#создание класса` лучше..тут код сам за себя говорит и такой комментарий только за зря на себя отбирает внимание

skyline-gleb commented on test_keys.md:
```diff
@@ -0,0 +1,17 @@
+- В поле "Откуда" набрать "Новосибирск" (достаточно набрать три первые буквы (регистр не важен) и затем в выпадающем списке можно кликнуть на нужный город)
+- В поле "Куда" набрать "Москва"
+- Поле "Когда" оставляем пустым.
+- Нажать на кнопку "найти".
```
вот не правда, шаг не "нажать на кнопку", а проверить что она отключена. То что человеку проверить отключенность проше попробовава нажать, это уже дело третье))
причем в тесте у тебя этот шаг описан верно - "и проверяем статус кнопки "Найти""

skyline-gleb commented on test_keys.md:
```diff
@@ -0,0 +1,17 @@
+- Нажать на кнопку "найти".
+
+### Expected Resul
+- Ничего не должно произойти. Запрос не должен выполняться.
```
и ожидаемый результат то что кнопка отключена, это более понятно и конкретно

skyline-gleb commented on main.py:
```diff
@@ -0,0 +1,44 @@
+        # заполняем два поля "откуда" и "куда"
+        # поле "когда" оставляем пустым
+        # и проверяем статус кнопки "Найти"
+        # надо, чтобы статус кнопки при этом был disabled
```
вот именно такого объединения кода и тест-кейса я ждал. Только совет, переоформи комментарий используя три двойных кавычки, чуть подробней и с примером тут https://www.python.org/dev/peps/pep-0008/#documentation-strings

skyline-gleb commented:
За оформление тест-кейса в md файл отдельный плюс, это действительно красиво и удобно читать. Но думаю сама понимаешь, при чтении автотеста важно когда описание как можно ближе к коду, то есть как ты сделала внутри теста.

lenabatik committed:
ispravila main.py po zame4anijam

lenabatik committed:
ispravila ozhidaemij rezul'tat v test-kejse

lenabatik committed:
Убрала лишние отступы

две строчки после изменения файла почему-то "отскочили" в сторону, хотя я соблюдала все отступы...
поэтому я снова отредактировала файл.

lenabatik commented on main.py:
```diff
@@ -0,0 +1,41 @@
+        поле "когда" оставляем пустым
+        и проверяем статус кнопки "Найти"
+        надо, чтобы статус кнопки при этом был disabled
+	"""
```
@skyline-gleb не понимаю, что тут с отступами? в 20-й строчке криво получилось, а в окне редактирования все ровно было.

skyline-gleb replied:
у тебя там табуляция вместо пробелов и как видишь вот результат, у тебя отображалось в одном виде, здесь в другом, у 3го человека ещё как-то. Исправляй на пробелы, это не единственное место у тебя с табуляцией.

lenabatik commented:
@skyline-gleb 
1. использовала метод button_status.is_enabled(), действительно красиво получилось, одна строчка вместо двух.
2. убрала лишний комментарий, который отвлекает на себя внимание.
3. исправила тест-кейс
4. переоформила комментарии как надо (по три двойные кавычки)

skyline-gleb commented on main.py:
```diff
@@ -0,0 +1,41 @@
+from unittest import TestCase
+from selenium import webdriver
+
+class Status_Button_Disabled_Test(TestCase):
```
не корректный нейминг.. имя класса правильно писать `StatusButtonDisabledTest`

lenabatik committed:
Исправление

Исправила название класса и два отступа

lenabatik commented:
@skyline-gleb уже исправила. и название класса, и отступы.

skyline-gleb commented:
принятно :+1:
